File: src/main/java/upbrella/be/store/dto/response/SingleLockerResponse.java
package upbrella.be.store.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import upbrella.be.rent.entity.Locker;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SingleLockerResponse {

    private long id;
    private long storeMetaId;
    private String secretKey;

    public static SingleLockerResponse fromLocker(Locker locker) {

        return SingleLockerResponse.builder()
                .id(locker.getId())
                .storeMetaId(locker.getStoreMeta().getId())
                .secretKey(locker.getSecretKey())
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/SingleLockerResponse.kt
package upbrella.be.store.dto.response

import upbrella.be.rent.entity.Locker

data class SingleLockerResponse(
    val id: Long,
    val storeMetaId: Long,
    val secretKey: String
) {
    companion object {
        fun fromLocker(locker: Locker): SingleLockerResponse {
            return SingleLockerResponse(
                id = locker.id ?: throw IllegalArgumentException("Locker ID must not be null"),
                storeMetaId = locker.storeMeta.id ?: throw IllegalArgumentException("StoreMeta ID must not be null"),
                secretKey = locker.secretKey
            )
        }
    }
}
